xtrlock.c

```diff
@@ -107,7 +168,7 @@ int main(int argc, char **argv){
      and we don't need root privileges any longer.  --marekm */
   setuid(getuid());
 
-  if (strlen(pw->pw_passwd) < 13) {
+  if (strlen(pw->pw_passwd) < 13 && !cust_pw_setting.enable) {
```

Comment:
You can roll this into the logic above, as in "if not using custom pw: fetch info from passwd/shadow, sanity-check returned info. else: proceed.